build/Dockerfile

```diff
@@ -1,26 +1,46 @@
-FROM centos:centos8
-
+FROM registry.access.redhat.com/ubi8/ubi:8.1
 # Use this build arg to set any default test script arguments
-ARG RUN_SCRIPT_ARGS=
 ENV RUN_SCRIPT_ARGS=${RUN_SCRIPT_ARGS}
-
+ENV ROBOT_EXTRA_ARGS=${ROBOT_EXTRA_ARGS}
 ENV HOME /tmp
 
 RUN dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm &&\
-    dnf install -y git python3 unzip chromium chromedriver &&\
+    dnf install -y git python3 unzip chromium chromedriver redhat-lsb-core &&\
     dnf clean all
 
+## Install yq in the image
+RUN curl -L https://github.com/mikefarah/yq/releases/download/v4.16.2/yq_linux_amd64 -o /usr/bin/yq &&\
+    chmod +x /usr/bin/yq
+
+## Install oc in the container
+COPY build/oc /usr/local/bin/oc
+# COPY oc /usr/local/bin/oc
+RUN chmod 755 /usr/local/bin/oc &&\
+         oc version --client &&\
+         oc config set-context loadtest
+
 RUN mkdir $HOME/ods-ci
 # Change the WORKDIR so the run script references any files/folders from the root of the repo
 WORKDIR $HOME/ods-ci
 
+COPY requirements.txt setup.py .
+RUN python3 -m venv venv && source venv/bin/activate &&  pip3 install --upgrade pip && venv/bin/pip3 install -r requirements.txt
+
 COPY tests tests/
-COPY requirements.txt run_robot_test.sh setup.py .
-RUN python3 -m venv venv && source venv/bin/activate && venv/bin/pip3 install -r requirements.txt
+COPY libs libs/
+COPY run_robot_test.sh  .
 
 # Set the group ownership so non-root users can write to /tmp
 RUN chgrp -R 0 /tmp && chmod -R g=u /tmp
+RUN chown -R 1001 /tmp
 
 USER 1001
 
-CMD ./run_robot_test.sh --skip-pip-install ${RUN_SCRIPT_ARGS}
+
+CMD if [ -n "${ROBOT_EXTRA_ARGS}" ]; then \
+            echo --extra-robot-args "${ROBOT_EXTRA_ARGS}" &&\
+	        ./run_robot_test.sh --skip-pip-install ${RUN_SCRIPT_ARGS} --extra-robot-args "${ROBOT_EXTRA_ARGS}" ;\
+     else \
+            echo execute all &&\
+            ./run_robot_test.sh --skip-pip-install ${RUN_SCRIPT_ARGS};\
+     fi
```

Comment:
Are the `echo` statements required?

Instead of embedding logic in the Dockerfile, you should just document that the user must include `--extra-robot-args` in the env var `ROBOT_EXTRA_ARGS`.  This would simplify this statement to
```
 ./run_robot_test.sh --skip-pip-install ${RUN_SCRIPT_ARGS} "${ROBOT_EXTRA_ARGS}"
```

Reply:
I can remove the echo statements, they were there for debugging while developing.

In origin, it was like you said, but it wasn't working as reported in a [comment](https://github.com/red-hat-data-services/ods-ci/pull/166#discussion_r782735501) in PR #166, that should be related to quotes escaping issue. In this way we workaround the issue.